netlib/utils.py

```diff
@@ -432,5 +432,26 @@ def safe_subn(pattern, repl, target, *args, **kwargs):
     """
     return re.subn(str(pattern), str(repl), target, *args, **kwargs)
 
-def bin_safe(s):
-    return ''.join(["\\x{:02x}".format(ord(i)) if ord(i) < 32 else i for i in s])
+
+def bytes_to_escaped_str(data):
+    """
+    Take bytes and return a safe string that can be displayed to the user.
+    """
+    if not isinstance(data, bytes):
+        raise ValueError("data must be bytes")
+    return repr(data).lstrip("b")[1:-1]
```

Comment:
This also encodes UTF8 multi-byte characters into hex. But we actually want those to remain untouched so the terminal renders them properly as what they are (`üäö` instead of some weird hex characters).
We only want to sanitise control characters (<32).